kafka_consumer/datadog_checks/kafka_consumer/client.py

```diff
@@ -104,12 +104,13 @@ def get_highwater_offsets(self, consumer_offsets):
     def get_partitions_for_topic(self, topic):
         try:
             cluster_metadata = self.kafka_client.list_topics(topic, timeout=self.config._request_timeout)
-            topic_metadata = cluster_metadata.topics[topic]
-            partitions = list(topic_metadata.partitions.keys())
-            return partitions
         except KafkaException as e:
             self.log.error("Received exception when getting partitions for topic %s: %s", topic, e)
             return None
+        else:
+            topic_metadata = cluster_metadata.topics[topic]
+            partitions = list(topic_metadata.partitions.keys())
+            return partitions
```

Comment:
We can leave this for later, but we don't really need the `else` block here anymore since the `except` clause already returns early. Removing the `else` makes it more clear what the main ("happy") path of the function is.